File: Brain Sticky/Core/Models/MindOSModels.swift
//
//  MindOSModels.swift
//  Brain Sticky
//
//  Created for Brain Sticky - Personal Second Brain Super App.
//

import Foundation
import SwiftUI

// MARK: - 极简双字分区
public enum OmniCategory: String, Codable, CaseIterable, Identifiable {
    case urgentTodo = "待办"
    case stickyNote = "日常"
    case passwordVault = "密码"
    case grocery = "买菜"
    case wishlist = "剁手"
    
    public var id: String { rawValue }
    
    public var icon: String {
        switch self {
        case .urgentTodo: return "checklist"
        case .stickyNote: return "sparkles"
        case .passwordVault: return "lock.fill"
        case .grocery: return "cart.fill"
        case .wishlist: return "gift.fill"
        }
    }
    
    public var themeColor: Color {
        switch self {
        case .urgentTodo: return BentoColors.urgentCoral
        case .stickyNote: return BentoColors.noteAmber
        case .passwordVault: return BentoColors.vaultViolet
        case .grocery: return BentoColors.groceryMint
        case .wishlist: return BentoColors.wishlistRuby
        }
    }
}

// MARK: - 1. 待办 (Todos)
public enum TodoPriority: String, Codable, CaseIterable {
    case urgent = "紧急"
    case normal = "日常"
    case someday = "随缘"
    
    public var color: Color {
        switch self {
        case .urgent: return BentoColors.urgentCoral
        case .normal: return BentoColors.noteAmber
        case .someday: return BentoColors.omniElectric
        }
    }
    
    public func localized(lang: AppLanguage = AppLanguageManager.shared.currentLanguage) -> String {
        switch self {
        case .urgent: return lang == .chinese ? "紧急" : "Urgent"
        case .normal: return lang == .chinese ? "日常" : "Normal"
        case .someday: return lang == .chinese ? "随缘" : "Someday"
        }
    }
}

public struct TodoItem: Identifiable, Codable, Hashable {
    public var id: UUID
    public var title: String
    public var isCompleted: Bool
    public var dueDate: Date?
    public var priority: TodoPriority
    public var reminderMinutes: Int?
    public var createdAt: Date
    
    public init(
        id: UUID = UUID(),
        title: String,
        isCompleted: Bool = false,
        dueDate: Date? = nil,
        priority: TodoPriority = .normal,
        reminderMinutes: Int? = nil,
        createdAt: Date = Date()
    ) {
        self.id = id
        self.title = title
        self.isCompleted = isCompleted
        self.dueDate = dueDate
        self.priority = priority
        self.reminderMinutes = reminderMinutes
        self.createdAt = createdAt
    }
}

// MARK: - 2. 点滴 (Moments & Drops)
public struct StickyNoteItem: Identifiable, Codable, Hashable {
    public var id: UUID
    public var content: String
    public var moodEmoji: String
    public var colorHex: String
    public var isPinned: Bool
    public var isEphemeral: Bool
    public var createdAt: Date
    
    public init(
        id: UUID = UUID(),
        content: String,
        moodEmoji: String = "💡",
        colorHex: String = "#FFF7D1",
        isPinned: Bool = false,
        isEphemeral: Bool = false,
        createdAt: Date = Date()
    ) {
        self.id = id
        self.content = content
        self.moodEmoji = moodEmoji
        self.colorHex = colorHex
        self.isPinned = isPinned
        self.isEphemeral = isEphemeral
        self.createdAt = createdAt
    }
}

// MARK: - 3. 密码 (Vault)
public enum VaultCategory: String, Codable, CaseIterable, Identifiable {
    case drop = "日常"
    case custom = "密码"
    
    public var id: String { rawValue }
    
    public var icon: String {
        switch self {
        case .drop: return "sparkles"
        case .custom: return "key.fill"
        }
    }
    
    public var defaultAccountLabel: String { "备注说明" }
    public var defaultSecretLabel: String { "密码" }
}

public struct VaultItem: Identifiable, Codable, Hashable {
    public var id: UUID
    public var title: String
    public var category: VaultCategory
    public var accountOrKey: String
    public var secretValue: String
    public var notes: String
    public var updatedAt: Date
    public var isMasked: Bool
    
    public init(
        id: UUID = UUID(),
        title: String,
        category: VaultCategory = .custom,
        accountOrKey: String = "",
        secretValue: String,
        notes: String = "",
        updatedAt: Date = Date(),
        isMasked: Bool = false
    ) {
        self.id = id
        self.title = title
        self.category = category
        self.accountOrKey = accountOrKey
        self.secretValue = secretValue
        self.notes = notes
        self.updatedAt = updatedAt
        self.isMasked = isMasked
    }
}

// MARK: - 4. 买菜 (Grocery)
public enum GroceryAisle: String, Codable, CaseIterable, Identifiable {
    case produce = "果蔬"
    case meat = "肉禽"
    case dairy = "乳品"
    case snacks = "零食"
    case essentials = "粮油"
    case daily = "百货"
    case others = "其他"
    
    public var id: String { rawValue }
    
    public var icon: String {
        switch self {
        case .produce: return "🥬"
        case .meat: return "🥩"
        case .dairy: return "🥛"
        case .snacks: return "🍿"
        case .essentials: return "🌾"
        case .daily: return "🧴"
        case .others: return "📦"
        }
    }
    
    public var cuteTitle: String {
        "\(icon) \(localized())"
    }
    
    public func localized(lang: AppLanguage = AppLanguageManager.shared.currentLanguage) -> String {
        switch self {
        case .produce: return lang == .chinese ? "果蔬" : "Produce"
        case .meat: return lang == .chinese ? "肉禽" : "Meat"
        case .dairy: return lang == .chinese ? "乳品" : "Dairy"
        case .snacks: return lang == .chinese ? "零食" : "Snacks"
        case .essentials: return lang == .chinese ? "粮油" : "Pantry"
        case .daily: return lang == .chinese ? "百货" : "Daily"
        case .others: return lang == .chinese ? "其他" : "Other"
        }
    }
}

public struct GroceryItem: Identifiable, Codable, Hashable {
    public var id: UUID
    public var name: String
    public var aisle: GroceryAisle
    public var quantity: String
    public var isBought: Bool
    public var isFrequent: Bool
    public var createdAt: Date
    
    public init(
        id: UUID = UUID(),
        name: String,
        aisle: GroceryAisle = .others,
        quantity: String = "1",
        isBought: Bool = false,
        isFrequent: Bool = false,
        createdAt: Date = Date()
    ) {
        self.id = id
        self.name = name
        self.aisle = aisle
        self.quantity = quantity
        self.isBought = isBought
        self.isFrequent = isFrequent
        self.createdAt = createdAt
    }
}

// MARK: - 5. 心愿 / 剁手 (Wishlist & Currencies)
public enum WishlistCurrency: String, Codable, CaseIterable, Identifiable {
    case cny = "¥" // 元
    case usd = "$" // 美元
    case jpy = "円" // 日币
    case eur = "€" // 欧元
    
    public var id: String { rawValue }
    
    public var name: String {
        switch self {
        case .cny: return AppLanguageManager.shared.currentLanguage == .chinese ? "元" : "CNY"
        case .usd: return AppLanguageManager.shared.currentLanguage == .chinese ? "美元" : "USD"
        case .jpy: return AppLanguageManager.shared.currentLanguage == .chinese ? "日币" : "JPY"
        case .eur: return AppLanguageManager.shared.currentLanguage == .chinese ? "欧元" : "EUR"
        }
    }
    
    public var symbol: String { rawValue }
    
    public func formatted(_ price: Double) -> String {
        if self == .jpy {
            return String(format: "%.0f 円", price)
        } else {
            return String(format: "%@%.0f", symbol, price)
        }
    }
    
    public static func fromSymbol(_ symbol: String) -> WishlistCurrency {
        switch symbol {
        case "$", "USD": return .usd
        case "円", "JPY", "JP¥": return .jpy
        case "€", "EUR": return .eur
        default: return .cny
        }
    }
}

public struct WishlistItem: Identifiable, Codable, Hashable {
    public var id: UUID
    public var title: String
    public var targetPrice: Double
    public var currency: String
    public var coolOffDaysTotal: Int
    public var coolOffStartDate: Date
    public var notes: String
    public var buyUrl: String
    public var isPurchased: Bool
    public var isAbandoned: Bool
    public var category: String
    
    public init(
        id: UUID = UUID(),
        title: String,
        targetPrice: Double = 0.0,
        currency: String = "¥",
        coolOffDaysTotal: Int = 14,
        coolOffStartDate: Date = Date(),
        notes: String = "",
        buyUrl: String = "",
        isPurchased: Bool = false,
        isAbandoned: Bool = false,
        category: String = "好物"
    ) {
        self.id = id
        self.title = title
        self.targetPrice = targetPrice
        self.currency = currency
        self.coolOffDaysTotal = coolOffDaysTotal
        self.coolOffStartDate = coolOffStartDate
        self.notes = notes
        self.buyUrl = buyUrl
        self.isPurchased = isPurchased
        self.isAbandoned = isAbandoned
        self.category = category
    }
    
    enum CodingKeys: String, CodingKey {
        case id, title, targetPrice, currency, coolOffDaysTotal, coolOffStartDate, notes, buyUrl, isPurchased, isAbandoned, category
    }
    
    public init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        self.id = try container.decodeIfPresent(UUID.self, forKey: .id) ?? UUID()
        self.title = try container.decodeIfPresent(String.self, forKey: .title) ?? ""
        self.targetPrice = try container.decodeIfPresent(Double.self, forKey: .targetPrice) ?? 0.0
        self.currency = try container.decodeIfPresent(String.self, forKey: .currency) ?? "¥"
        self.coolOffDaysTotal = try container.decodeIfPresent(Int.self, forKey: .coolOffDaysTotal) ?? 14
        self.coolOffStartDate = try container.decodeIfPresent(Date.self, forKey: .coolOffStartDate) ?? Date()
        self.notes = try container.decodeIfPresent(String.self, forKey: .notes) ?? ""
        self.buyUrl = try container.decodeIfPresent(String.self, forKey: .buyUrl) ?? ""
        self.isPurchased = try container.decodeIfPresent(Bool.self, forKey: .isPurchased) ?? false
        self.isAbandoned = try container.decodeIfPresent(Bool.self, forKey: .isAbandoned) ?? false
        self.category = try container.decodeIfPresent(String.self, forKey: .category) ?? "好物"
    }
    
    public var daysRemaining: Int {
        if coolOffDaysTotal >= 9999 { return 9999 }
        let calendar = Calendar.current
        let elapsed = calendar.dateComponents([.day], from: coolOffStartDate, to: Date()).day ?? 0
        let remaining = coolOffDaysTotal - elapsed
        return max(0, remaining)
    }
    
    public var coolOffProgress: Double {
        guard coolOffDaysTotal > 0 else { return 1.0 }
        let calendar = Calendar.current
        let elapsed = calendar.dateComponents([.day], from: coolOffStartDate, to: Date()).day ?? 0
        return min(1.0, max(0.0, Double(elapsed) / Double(coolOffDaysTotal)))
    }
}

// MARK: - 6. 自定义功能模块 (Custom Modular Cards)
public struct CustomEntryItem: Identifiable, Codable, Hashable {
    public var id: UUID
    public var title: String
    public var icon: String // 每个打卡项独立的专属 Icon
    public var detail: String
    public var isCompleted: Bool
    public var targetDate: Date?
    public var count: Int
    public var createdAt: Date
    
    public init(
        id: UUID = UUID(),
        title: String,
        icon: String = "🎯",
        detail: String = "",
        isCompleted: Bool = false,
        targetDate: Date? = nil,
        count: Int = 0,
        createdAt: Date = Date()
    ) {
        self.id = id
        self.title = title
        self.icon = (icon.isEmpty || icon == "🎯") ? CustomEntryItem.suggestIcon(for: title) : icon
        self.detail = detail
        self.isCompleted = isCompleted
        self.targetDate = targetDate
        self.count = count
        self.createdAt = createdAt
    }
    
    enum CodingKeys: String, CodingKey {
        case id, title, icon, detail, isCompleted, targetDate, count, createdAt
    }
    
    public init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        self.id = try container.decodeIfPresent(UUID.self, forKey: .id) ?? UUID()
        self.title = try container.decodeIfPresent(String.self, forKey: .title) ?? ""
        let decodedIcon = try container.decodeIfPresent(String.self, forKey: .icon)
        self.icon = (decodedIcon == nil || decodedIcon!.isEmpty) ? CustomEntryItem.suggestIcon(for: self.title) : decodedIcon!
        self.detail = try container.decodeIfPresent(String.self, forKey: .detail) ?? ""
        self.isCompleted = try container.decodeIfPresent(Bool.self, forKey: .isCompleted) ?? false
        self.targetDate = try container.decodeIfPresent(Date.self, forKey: .targetDate)
        self.count = try container.decodeIfPresent(Int.self, forKey: .count) ?? 0
        self.createdAt = try container.decodeIfPresent(Date.self, forKey: .createdAt) ?? Date()
    }
    
    public static func suggestIcon(for text: String) -> String {
        let lower = text.lowercased()
        if lower.contains("水") || lower.contains("water") || lower.contains("drink") { return "💧" }
        if lower.contains("书") || lower.contains("读") || lower.contains("read") || lower.contains("book") { return "📖" }
        if lower.contains("跑") || lower.contains("run") || lower.contains("运动") || lower.contains("练") || lower.contains("gym") || lower.contains("健身") { return "🏃" }
        if lower.contains("睡") || lower.contains("sleep") || lower.contains("晚安") || lower.contains("早起") || lower.contains("起") { return "🌙" }
        if lower.contains("多邻国") || lower.contains("duolingo") || lower.contains("鸟") { return "🦉" }
        if lower.contains("词") || lower.contains("背") || lower.contains("学") || lower.contains("study") || lower.contains("英") || lower.contains("语") { return "🦉" }
        if lower.contains("药") || lower.contains("维生素") || lower.contains("pill") { return "💊" }
        if lower.contains("咖啡") || lower.contains("茶") || lower.contains("tea") || lower.contains("coffee") { return "☕️" }
        if lower.contains("钱") || lower.contains("存") || lower.contains("薪") || lower.contains("pay") || lower.contains("理财") { return "💰" }
        if lower.contains("娃") || lower.contains("宝") || lower.contains("儿") || lower.contains("孩") || lower.contains("baby") || lower.contains("kid") { return "👶" }
        if lower.contains("猫") || lower.contains("狗") || lower.contains("宠") || lower.contains("pet") { return "🐾" }
        if lower.contains("生") || lower.contains("日") || lower.contains("birthday") || lower.contains("party") { return "🎂" }
        if lower.contains("冥想") || lower.contains("放") || lower.contains("瑜伽") { return "🧘" }
        if lower.contains("家") || lower.contains("饭") || lower.contains("做饭") || lower.contains("厨") || lower.contains("餐") { return "🍳" }
        if lower.contains("果") || lower.contains("蔬") || lower.contains("沙拉") || lower.contains("eat") { return "🍎" }
        if lower.contains("琴") || lower.contains("乐") || lower.contains("歌") { return "🎸" }
        if lower.contains("行") || lower.contains("飞") || lower.contains("游") || lower.contains("travel") { return "✈️" }
        if lower.contains("恋") || lower.contains("爱") || lower.contains("纪念") || lower.contains("love") { return "❤️" }
        if lower.contains("洁") || lower.contains("扫") || lower.contains("家务") || lower.contains("整") { return "🧹" }
        if lower.contains("山") || lower.contains("海") || lower.contains("爬") || lower.contains("锻炼") { return "⛰️" }
        if lower.contains("骑") || lower.contains("车") || lower.contains("bike") { return "🚴" }
        return "🎯"
    }
    
    public var daysUntilTarget: Int? {
        guard let targetDate = targetDate else { return nil }
        let calendar = Calendar.current
        let start = calendar.startOfDay(for: Date())
        let end = calendar.startOfDay(for: targetDate)
        return calendar.dateComponents([.day], from: start, to: end).day
    }
}

public struct CustomModuleCard: Identifiable, Codable, Hashable {
    public var id: String // "custom_1", "custom_2"
    public var title: String // e.g. "打卡", "倒数"
    public var icon: String // "🎯", "⏳", "💧", "📚", etc.
    public var colorHex: String
    public var entries: [CustomEntryItem]
    public var mode: String // "checkin" | "countdown" | "general"
    
    public init(
        id: String,
        title: String,
        icon: String = "🎯",
        colorHex: String = "#E0F2FE",
        entries: [CustomEntryItem] = [],
        mode: String = "checkin"
    ) {
        self.id = id
        self.title = title
        self.icon = icon
        self.colorHex = colorHex
        self.entries = entries
        self.mode = mode
    }
}

public struct CustomPresetItem: Identifiable, Codable, Hashable {
    public var id: UUID
    public var icon: String
    public var name: String
    public var label: String
    public var defaultDetail: String
    
    public init(id: UUID = UUID(), icon: String, name: String, label: String, defaultDetail: String = "") {
        self.id = id
        self.icon = icon
        self.name = name
        self.label = label
        self.defaultDetail = defaultDetail
    }
}

File: app/src/main/java/com/example/brainsticky/model/MindOSModels.kt
package com.example.brainsticky.model

import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
enum class AppLanguage(val code: String, val displayName: String) {
    CHINESE("zh", "🇨🇳 简体中文"),
    ENGLISH("en", "🇺🇸 English")
}

@Serializable
enum class BentoCategory(val titleZh: String, val titleEn: String, val icon: String, val colorHex: String) {
    TODO("待办", "Todo", "checklist", "#FF6B6B"),
    DROPS("日常", "Daily", "sparkles", "#FF9F1C"),
    VAULT("密码", "Vault", "lock", "#9D4EDD"),
    GROCERY("买菜", "Market", "cart", "#2EC4B6"),
    WISHLIST("剁手", "Wishlist", "gift", "#FF4081"),
    HABIT("打卡", "Check-in", "target", "#4D88FF")
}

enum class OmniType(val titleZh: String, val titleEn: String) {
    TODO("待办", "Todo"),
    DROP("日常", "Daily"),
    GROCERY("买菜", "Market"),
    WISHLIST("剁手", "Wishlist"),
    PASSWORD("密码", "Vault"),
    HABIT("打卡", "Check-in")
}

@Serializable
enum class TodoPriority(val labelZh: String, val labelEn: String, val hexColor: String) {
    URGENT("紧急", "Urgent", "#FF5A5F"),
    NORMAL("日常", "Normal", "#FF9F1C"),
    SOMEDAY("随缘", "Someday", "#4D88FF");

    fun getLabel(lang: AppLanguage): String = if (lang == AppLanguage.CHINESE) labelZh else labelEn
}

@Serializable
data class TodoItem(
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val isCompleted: Boolean = false,
    val priority: TodoPriority = TodoPriority.NORMAL,
    val reminderMinutes: Int? = null,
    val createdAt: Long = System.currentTimeMillis()
)

@Serializable
data class StickyNoteItem(
    val id: String = UUID.randomUUID().toString(),
    val content: String,
    val moodEmoji: String = "✨",
    val colorHex: String = "#FFF7D1",
    val textColorHex: String = "#1E293B",
    val isPinned: Boolean = false,
    val createdAt: Long = System.currentTimeMillis()
)

@Serializable
enum class VaultCategory(val labelZh: String, val labelEn: String) {
    DROP("日常", "Drops"),
    CUSTOM("密码", "Custom");

    fun getLabel(lang: AppLanguage): String = if (lang == AppLanguage.CHINESE) labelZh else labelEn
}

@Serializable
data class VaultItem(
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val category: VaultCategory = VaultCategory.CUSTOM,
    val accountOrKey: String = "",
    val secretValue: String,
    val notes: String = "",
    val updatedAt: Long = System.currentTimeMillis(),
    val isMasked: Boolean = true
)

@Serializable
enum class GroceryAisle(val labelZh: String, val labelEn: String, val icon: String) {
    PRODUCE("果蔬", "Produce", "🥬"),
    MEAT("肉禽", "Meat", "🥩"),
    DAIRY("乳品", "Dairy", "🥛"),
    SNACKS("零食", "Snacks", "🍿"),
    ESSENTIALS("粮油", "Pantry", "🌾"),
    DAILY("百货", "Daily", "🧴"),
    OTHERS("其他", "Other", "📦");

    fun getLabel(lang: AppLanguage): String = if (lang == AppLanguage.CHINESE) labelZh else labelEn
    fun getCuteTitle(lang: AppLanguage): String = "$icon ${getLabel(lang)}"
}

@Serializable
data class GroceryItem(
    val id: String = UUID.randomUUID().toString(),
    val name: String,
    val aisle: GroceryAisle = GroceryAisle.OTHERS,
    val quantity: String = "1",
    val isBought: Boolean = false,
    val isFrequent: Boolean = false,
    val createdAt: Long = System.currentTimeMillis()
)

@Serializable
enum class WishlistCurrency(val symbol: String, val nameZh: String, val nameEn: String) {
    CNY("¥", "元", "CNY"),
    USD("$", "美元", "USD"),
    JPY("円", "日币", "JPY"),
    EUR("€", "欧元", "EUR");

    fun getName(lang: AppLanguage): String = if (lang == AppLanguage.CHINESE) nameZh else nameEn

    fun format(price: Double): String {
        return if (this == JPY) {
            if (price < 0) "-${kotlin.math.abs(price.toLong())} 円" else "${price.toLong()} 円"
        } else {
            if (price < 0) "-$symbol${kotlin.math.abs(price.toLong())}" else "$symbol${price.toLong()}"
        }
    }

    companion object {
        fun fromSymbol(sym: String): WishlistCurrency {
            return when (sym) {
                "$", "USD" -> USD
                "円", "JPY", "JP¥" -> JPY
                "€", "EUR" -> EUR
                else -> CNY
            }
        }
    }
}

@Serializable
data class WishlistItem(
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val targetPrice: Double = 0.0,
    val currency: String = "¥",
    val coolOffDaysTotal: Int = 14,
    val coolOffStartDate: Long = System.currentTimeMillis(),
    val notes: String = "",
    val isPurchased: Boolean = false,
    val isAbandoned: Boolean = false
) {
    val daysPassed: Int
        get() {
            val diff = System.currentTimeMillis() - coolOffStartDate
            return (diff / (1000 * 60 * 60 * 24)).toInt().coerceAtLeast(0)
        }

    val daysRemaining: Int
        get() = if (coolOffDaysTotal >= 9999) 9999 else (coolOffDaysTotal - daysPassed).coerceAtLeast(0)

    val coolOffProgress: Float
        get() {
            if (coolOffDaysTotal >= 9999) return 0f
            if (coolOffDaysTotal <= 0) return 1f
            return (daysPassed.toFloat() / coolOffDaysTotal.toFloat()).coerceIn(0f, 1f)
        }
}

@Serializable
data class CustomEntryItem(
    val id: String = UUID.randomUUID().toString(),
    val icon: String = "⭐️",
    val title: String,
    val detail: String = "",
    val isCompleted: Boolean = false,
    val streakDays: Int = 0,
    val count: Int = 0,
    val lastCompletedDate: String = "",
    val historyDates: List<String> = emptyList(),
    val lastCheckedInTimestamp: Long = 0L
) {
    val isCheckedInWithin24Hours: Boolean
        get() {
            if (lastCheckedInTimestamp <= 0L) return count > 0
            return (System.currentTimeMillis() - lastCheckedInTimestamp) < 24L * 3600L * 1000L
        }

    fun nextCheckInCountdown(lang: AppLanguage): String {
        if (lastCheckedInTimestamp <= 0L) {
            return if (lang == AppLanguage.CHINESE) "24小时后可再次打卡" else "Next check-in in 24h"
        }
        val elapsed = System.currentTimeMillis() - lastCheckedInTimestamp
        val total24h = 24L * 3600L * 1000L
        if (elapsed >= total24h) {
            return if (lang == AppLanguage.CHINESE) "已满24小时，可+1打卡 ✨" else "24h reached, ready for +1 ✨"
        }
        val remaining = total24h - elapsed
        val hours = (remaining / (1000 * 3600)).toInt()
        val minutes = ((remaining % (1000 * 3600)) / (1000 * 60)).toInt()
        return if (lang == AppLanguage.CHINESE) {
            if (hours > 0) "24小时倒数：${hours}小时${minutes}分后可再次打卡" else "24小时倒数：${kotlin.math.max(1, minutes)}分钟后可再次打卡"
        } else {
            if (hours > 0) "Next +1 available in ${hours}h ${minutes}m" else "Next +1 available in ${kotlin.math.max(1, minutes)}m"
        }
    }
}

@Serializable
data class CustomModule(
    val id: String = "habit_module",
    val title: String = "打卡",
    val subtitle: String = "坚持微小日常，日积月累 ✨",
    val icon: String = "target",
    val themeColorHex: String = "#4D88FF",
    val entries: List<CustomEntryItem> = emptyList()
)

data class BuiltinHabitPreset(
    val icon: String,
    val titleZh: String,
    val titleEn: String,
    val detailZh: String,
    val detailEn: String
) {
    fun getTitle(lang: AppLanguage): String = if (lang == AppLanguage.CHINESE) titleZh else titleEn
    fun getDetail(lang: AppLanguage): String = if (lang == AppLanguage.CHINESE) detailZh else detailEn

    companion object {
        val ALL = listOf(
            BuiltinHabitPreset("🏃", "晨跑锻炼", "Morning Run", "有氧 20 分钟", "20 mins cardio"),
            BuiltinHabitPreset("💧", "多喝水", "Drink Water", "每天 8 杯温水", "8 cups of water"),
            BuiltinHabitPreset("🌙", "早睡早起", "Sleep Early", "23:00 前放下手机", "No phone after 11 PM"),
            BuiltinHabitPreset("📖", "每日阅读", "Daily Reading", "读 10 页好书", "Read 10 pages"),
            BuiltinHabitPreset("🧘", "正念冥想", "Meditation", "深呼吸放空 10 分钟", "10 mins breathing"),
            BuiltinHabitPreset("🇬🇧", "背单词", "Vocab & English", "打卡 20 个新单词", "Learn 20 new words"),
            BuiltinHabitPreset("💰", "今日记账", "Expense Tracking", "记录每一笔开销", "Log all daily spendings"),
            BuiltinHabitPreset("🧴", "早晚护肤", "Skincare Routine", "防晒与补水", "Moisturize & sunscreen"),
            BuiltinHabitPreset("✨", "其他", "Other", "", "")
        )
    }
}
